File: c/4th_Sem_Lab/Q2.c
// a^b
#include <stdio.h>
int pow(int a,int b){
    if(b==0) return 1;
    if(b==1) return a;
    return a*pow(a,b-1);
}
int main(){
    int a,b;
    printf("Enter two numbers : ");
    scanf("%d %d",&a,&b);
    printf("%d to the power %d is %d\n",a,b,pow(a,b));
}

File: java/Lab_Assignment_1/Introduction_Programs/Q2.java
// Write a java program to compute the GCD of two numbers.

import java.util.Scanner;

public class Q2 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter two numbers : ");
        int x = sc.nextInt();
        int y = sc.nextInt();

        GCD obj = new GCD(x,y);

        System.out.println("The GCD of "+x+" and "+y+" is "+obj.calc());
        
        sc.close();
    }
}

class GCD{
    private int a;
    private int b;

    GCD(int a, int b){
        this.a = Math.max(a, b);
        this.b = Math.min(a, b);
    }

    int calc(){
        int rem = a % b;
        while(rem!=0){
            a = b;
            b = rem;
            rem = a % b;
        }
        return b;
    }
}